### PR TITLE
[ch1568] sections management

### PR DESCRIPTION
**Pull request by domdinicola**

### CODE REVIEW

File: src/etools/applications/management/migrations/0004_auto_20190715_2047.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Generated by Django 2.2.3 on 2019-07-15 20:47
```
Comment (reinbach) [inline src/etools/applications/management/migrations/0004_auto_20190715_2047.py]: Any reason not to merge this migration with the previous one? Minimize the number of migrations that need to run

File: src/etools/applications/management/views/sections.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            'pk': section.pk,
+            'name': section.name,
+        })
+
+    @action(detail=False, methods=['post'])
+    def merge(self, request):
+        try:
+            new_section_name = request.data['new_section_name']
+            sections_to_merge = request.data['sections_to_merge']
+            print('Section to Merge', sections_to_merge)
```
Comment (reinbach) [inline src/etools/applications/management/views/sections.py]: Is the `print` statement still needed?

File: src/etools/applications/management/views/sections.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            'name': section.name,
+        })
+
+    @action(detail=False, methods=['post'])
+    def close(self, request):
+        try:
+            old_section = request.data['old_section']
+            objects_dict = request.data['new_sections']
+
+            print('Section Migrate', objects_dict)
```
Comment (reinbach) [inline src/etools/applications/management/views/sections.py]: Is the `print` statement still needed?

File: src/etools/applications/management/migrations/0004_auto_20190715_2047.py (hunk 1 above)
Comment (domdinicola) [reply src/etools/applications/management/migrations/0004_auto_20190715_2047.py]: yeah once approved, I'll do that a squash the commits I guess.
I will coordinate with @marko911 cause I don't want him to throw away his db

**Comment by domdinicola** [comment]
[ch1568]

File: src/etools/applications/management/migrations_issues.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+from etools.applications.core.util_scripts import set_country
+from etools.applications.reports.models import Indicator, Result
+from etools.applications.users.models import Country
```
Comment (robertavram) [inline src/etools/applications/management/migrations_issues.py]: please rm from codebase

File: src/etools/applications/management/handlers.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        section_history.to_sections.set([to_instance, ])
+
+        return to_instance
+
+    @staticmethod
+    @atomic
+    def close(from_instance_pk, new_section_2_new_querysets):
+        """
+        Close a section and split the active objects according the mapping dictionary
+        new_section_2_new_querysets has a mapping of the instances to be mapped to new section
```
Comment (robertavram) [inline src/etools/applications/management/handlers.py]: update comment

File: src/etools/applications/management/views/sections.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        try:
+            new_section_name = request.data['new_section_name']
+            sections_to_merge = request.data['sections_to_merge']
+            logger.info('Section to Merge', sections_to_merge)
+        except KeyError:
+            return Response(_('Unable to unpack'), status=status.HTTP_400_BAD_REQUEST)
+        try:
+            section = SectionHandler.merge(new_section_name, sections_to_merge)
+        except IntegrityError:
+            return Response(f'Section {new_section_name} already exist', status=status.HTTP_400_BAD_REQUEST)
```
Comment (robertavram) [inline src/etools/applications/management/views/sections.py]: err handling

File: src/etools/applications/partners/serializers/interventions_v2.py [hunk 7]
```diff
@@ -265,6 +266,27 @@ class Meta:
         )
 
 
+class BasicInterventionListSerializer(serializers.ModelSerializer):
```
Comment (robertavram) [inline src/etools/applications/partners/serializers/interventions_v2.py]: improve naming

**Pull request merged**